### ISSUE TITLE
Delete Comment is not working expected

### ISSUE THREAD

**Issue by surapuramakhil** [apache/incubator-answer#796]
## Describe the bug

I have made an test comment from my account akhil_test. I have tagged another acount akhil in the comment.

in my akhil_test account (person who wrote the comment) has deleted the comment. The comment is only deleted for him but his comment still exists in akhil account. and in public internet.

Following screenshots are taken at same time. Testing on https://meta.answer.dev/
place where its tested : https://meta.answer.dev/questions/D1eI2/?commentId=10070000000010357

From akhil_test account
![Screenshot (231)](https://github.com/apache/incubator-answer/assets/9161543/2683cd4c-12a2-4320-943f-0c6ae7ec17be)

public internet (No user loggedIn)
![Screenshot (230)](https://github.com/apache/incubator-answer/assets/9161543/bd0dd9e6-6614-4607-8330-d6df37a55c4a)

From account Akhil
![Screenshot (229)](https://github.com/apache/incubator-answer/assets/9161543/f0c9149b-1a3f-4334-a98f-95bf37f2316d)

### Screenshots

If applicable, add screenshots or video to help explain your problem.

### Platform

- Device: [e.g. Desktop, Mobile]
- OS: [e.g. macOS]
- Browser and version: [e.g. Chrome, Safari]
- Version: [e.g. v1.2.0]

**Comment by hgaol** [comment]
have you refreshed the page?

**Comment by surapuramakhil** [comment]
yes multiple times. Its still there. Just now I checked the page.

**Comment by hgaol** [comment]
I found that if visiting your link which pointed to the comment (https://meta.answer.dev/questions/D1eI2/?commentId=10070000000010357), the comment will stay there. If I visit this question directly (https://meta.answer.dev/questions/D1eI2), it disappeared. Not sure if it's by design.

**Comment by zahash** [comment]
I made a PR to fix this issue. https://github.com/apache/incubator-answer/pull/799
can one of the maintainers please look into this?

**Comment by zahash** [comment]
@LinkinStars

**Comment by surapuramakhil** [comment]
Thank you, @LinkinStars can you consider adding e2e test case for this.

**Comment by zahash** [comment]
oh didn't realize this issue is still open. sure i will add the two test cases.

**Comment by zahash** [comment]
@fenbox how do i fork the dev branch?

**Comment by surapuramakhil** [comment]
@zahash create a branch from your project dev branch. Make sure you sync your project before creating a branch

**Comment by zahash** [comment]
@LinkinStars 

how do i run tests? i get this error when i `make test`

```sh
try to create database directory /
[xorm] [info]  2024/03/21 17:25:05.478650 PING DATABASE sqlite
panic: connection to database failed: unable to open database file: out of memory (14)

goroutine 1 [running]:
github.com/apache/incubator-answer/internal/repo/repo_test_test.TestMain(0xc001209cc0)
        /home/zahash/source/incubator-answer/internal/repo/repo_test/repo_main_test.go:88 +0x255
main.main()
        _testmain.go:165 +0x195
FAIL    github.com/apache/incubator-answer/internal/repo/repo_test      0.037s
FAIL
make: *** [Makefile:32: test] Error 1
```


btw this is what i do for launching the project
```sh
make ui
make build
sudo ./answer run --data-path ./data/
```

**Comment by surapuramakhil** [comment]
Are you out of memory? Probably, you can use CI/CD of GitHub for executing test cases. They offer unlimited for open source projects. 
you can trigger them by raising MR. provided this repo this configured for running test suite.

**Comment by zahash** [comment]
its not because of memory. for some reason the project is trying to create a temporary database at root. so obviously it doesn't have the permissions

`/tmpanswer-test-data.db`

```go
func TestMain(t *testing.M) {
	dbSetting, ok := dbSettingMapping[os.Getenv("TEST_DB_DRIVER")]
	if !ok {
		// Use sqlite3 to test.
		dbSetting = dbSettingMapping[string(schemas.SQLITE)]      // <<<<<<<<<<<<< HERE
	}
	if dbSetting.Driver == string(schemas.SQLITE) {
		os.RemoveAll(dbSetting.Connection)
	}

	defer func() {
		if tearDown != nil {
			tearDown()
		}
	}()
	if err := initTestDataSource(dbSetting); err != nil {
		panic(err)
	}
	log.Info("init test database successfully")

	if ret := t.Run(); ret != 0 {
		panic(ret)
	}
}
```

**Comment by zahash** [comment]
@LinkinStars i found the bug.

there should be a slash between temp dir and the db name

```go
	sqlite3DBSetting = TestDBSetting{
		Driver:     string(schemas.SQLITE),
		Connection: os.TempDir() + "answer-test-data.db",   // <<<<<< HERE
	}
```

right now, instead of creating a database at `/tmp/answer-test-data.db` it instead tries to create `/tmpanswer-test-data.db`

**Comment by surapuramakhil** [comment]
@zahash these test cases were created for container environment. It's not a bug, it's actually enhancement

**Comment by zahash** [comment]
> @zahash these test cases were created for container environment. It's not a bug, it's actually enhancement

no i don't think so. if it was created for container environment, why not just do `Connection: "answer-test-data.db"`

they clearly tried to put it in temp dir but made a mistake joining the paths

**Comment by surapuramakhil** [comment]
Possible hacks :joy:

**Comment by zahash** [comment]
@LinkinStars @fenbox here is the PR that adds tests to make sure deleted comments cannot be retrieved #872

and another PR to fix the filepath bug #871

**Comment by zahash** [comment]
@surapuramakhil

eh, mistakes like this happen from time to time.

but it is a little strange that Go thinks filepaths and filenames are utf8 strings.
because in most operating systems they can be arbitrary byte sequences separated by slashes.

**Comment by zahash** [comment]
@LinkinStars @surapuramakhil @fenbox can this issue be closed now that all the PRs have been merged? #872 #871
